src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -91,13 +90,14 @@ public class SqlQueryEngine : IQueryEngine
         /// Executes the given IMiddlewareContext of the GraphQL and expecting result of stored-procedure execution as
         /// list of Jsons and the relevant pagination metadata back.
         /// </summary>
-        public async Task<Tuple<IEnumerable<JsonDocument>, IMetadata?>> ExecuteListAsync(IMiddlewareContext context, IDictionary<string, object?> parameters, string dataSourceName = "")
+        public async Task<Tuple<IEnumerable<JsonDocument>, IMetadata?>> ExecuteListAsync(IMiddlewareContext context, IDictionary<string, object?> parameters, string dataSourceName)
```

Comment:
Even for SQL, dont we need to set to default data source if its empty?

Reply:
all callers of this function are explicitly setting the datasource name (if it is default like in rest case it is derived there only). Hence I did not see the need to make this a default parameter.